packages/block-editor/src/store/private-selectors.js

```diff
@@ -480,18 +481,65 @@ export const getContentLockingParent = createSelector(
 		let current = clientId;
 		let result;
 		while ( ( current = state.blocks.parents.get( current ) ) ) {
-			if (
-				getBlockName( state, current ) === 'core/block' ||
-				getTemplateLock( state, current ) === 'contentOnly'
-			) {
+			if ( getTemplateLock( state, current ) === 'contentOnly' ) {
+				result = current;
+			}
+		}
+		return result;
+	},
+	( state ) => [
+		state.blocks.parents,
+		state.blockListSettings,
+		state.editorMode,
+		getSectionRootClientId( state ),
+	]
```

Comment:
At a glance it looks like there are too many dependencies. I can't see that `getTemplateLock` depends on edit mode or the `sectionRootClientId`.

Reply:
yep, I split two selectors and kept the original dependencies, I'll fix it.